csharp/src/Google.Protobuf/WritingPrimitives.cs

```diff
@@ -163,6 +164,38 @@ public static void WriteBool(ref Span<byte> buffer, ref WriterInternalState stat
         /// </summary>
         public static void WriteString(ref Span<byte> buffer, ref WriterInternalState state, string value)
         {
+#if !NETSTANDARD1_1
+            const int MaxBytesPerChar = 3;
```

Comment:
drive by - why is the max 3?

Reply:
Most Chars are represented by one or two bytes. The maximum is 3. That's just how UTF8 works.

Reply:
That's what I thought you were going for, but then wouldn't the answer be 4 bytes not 3? https://tools.ietf.org/html/rfc3629 -

```
Char. number range  |        UTF-8 octet sequence
      (hexadecimal)    |              (binary)
   --------------------+---------------------------------------------
   0000 0000-0000 007F | 0xxxxxxx
   0000 0080-0000 07FF | 110xxxxx 10xxxxxx
   0000 0800-0000 FFFF | 1110xxxx 10xxxxxx 10xxxxxx
   0001 0000-0010 FFFF | 11110xxx 10xxxxxx 10xxxxxx 10xxxxxx
```

Depending on what the initial string encoding/length is (UTF16?), you might also have to look out for the surrogate pairs that confuse the issue.

Reply:
https://source.dot.net/#System.Private.CoreLib/UTF8Encoding.cs,49

Reply:
Thanks!